### PR TITLE
feat(raster)!: update titiler-pgstac from 0.2.3 to 0.8.0

### PR DESCRIPTION
**Pull request by vincentsarago**
ref #234 

This PR aims to update titiler-pgstac from version 0.2.3 to 0.8.0. Doing so, will also update titiler, rio-tiler, morecantile, fastapi and pydantic.

Endpoint changes 

- ~~Use `/collections/{collection_id}/items/{item_id}` prefix for Item endpoint.~~

I was able to keep this as it was (`{endpoint}/stac/info?collection=collection1&item=item1`)

- `post_process=` -> `algorithm=`

- change tile url path parameter order from `/tiles/{searchid}/{TileMatrixSetId}/{z}/{x}/{y`} to `/{searchid}/tiles/{TileMatrixSetId}/{z}/{x}/{y}`

```
# Before
{endpoint}/mosaic/tiles/20200307aC0853900w361030/0/0/0

# Now
{endpoint}/mosaic/20200307aC0853900w361030/tiles/0/0/0
```

- replace `/{searchid}/{z}/{x}/{y}/assets` endpoints by `/{searchid}/tiles/{z}/{x}/{y}/assets`

- replace - by _ in query parameters

    - coord-crs -> coord_crs
    - dst-crs -> dst_crs

- remove `max_size` default for mosaic's `/statistics [POST]` endpoint breaking change

- add `/bbox` and `/feature [POST]` optional endpoints

### CODE REVIEW

File: raster_api/runtime/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM public.ecr.aws/sam/build-python3.9:latest
+FROM public.ecr.aws/sam/build-python3.11:latest
```
Comment (vincentsarago) [inline raster_api/runtime/Dockerfile:1]: not mandatory but will be faster

File: raster_api/runtime/src/app.py [hunk 2]
```diff
@@ -1,124 +1,114 @@
...
 stac = MultiBaseTilerFactory(
     reader=PgSTACReader,
     path_dependency=ItemPathParams,
     optional_headers=optional_headers,
     router_prefix=f"{path_prefix}/stac",
     environment_dependency=settings.get_gdal_config,
     router=APIRouter(route_class=LoggerRouteHandler),
+    extensions=[
+        stacViewerExtension(),
+    ],
```
Comment (vincentsarago) [inline raster_api/runtime/src/app.py:92]: We now use `extensions` instead of creating custom Factories

File: raster_api/runtime/src/app.py [hunk 3]
```diff
@@ -1,124 +1,114 @@
...
+###############################################################################
 cog = TilerFactory(
     router_prefix=f"{path_prefix}/cog",
     optional_headers=optional_headers,
     environment_dependency=settings.get_gdal_config,
     router=APIRouter(route_class=LoggerRouteHandler),
+    extensions=[
+        cogValidateExtension(),
+        cogViewerExtension(),
+    ],
```
Comment (vincentsarago) [inline raster_api/runtime/src/app.py:109]: Using the official titiler extensions for the `/validate` and `/viewer` endpoints

File: raster_api/runtime/src/dependencies.py [hunk 4]
```diff
@@ -1,76 +1,23 @@
...
-            features = resp.get("features", [])
-            if not len(features):
-                raise Exception(
-                    "No item '{item_id}' found in '{collection_id}' collection"
-                )
-
-            return features[0]
-
-
-def DatasetPathParams(
```
Comment (vincentsarago) [inline raster_api/runtime/src/dependencies.py:54]: I don't think this was used

File: raster_api/runtime/setup.py [hunk 5]
```diff
@@ -6,12 +6,12 @@
     long_description = f.read()
 
 inst_reqs = [
-    "titiler.pgstac==0.2.3",
-    "titiler.application>=0.10,<0.11",
-    "importlib_resources>=1.1.0;python_version<='3.9'",  # https://github.com/cogeotiff/rio-tiler/pull/379
+    "boto3",
```
Comment (vincentsarago) [inline raster_api/runtime/setup.py:9]: `boto3` is now an optional dependency in rio-tiler

**Review by anayeaye — APPROVED** [review]

**Pull request merged**